simtool.cc

```diff
@@ -7150,6 +7150,92 @@ const char *tool_stop_mover_t::do_work( player_t *player, const koord3d &last_po
 	return NULL;
 }
 
+bool tool_change_city_of_building_t::init(player_t* player) {
+	two_click_tool_t::init(player);
+
+	// convert default_param to city and update highlight_city
+	highlight_city = welt->get_cities()[atoi(default_param)];
```

Comment:
Could you check whether the tool works with a network game?
`default_param` is usually set by the caller of the toll, using `set_default_param` function.